src/v/rpc/connection_cache.cc

```diff
@@ -88,4 +99,54 @@ ss::future<> connection_cache::stop() {
     return _gate.close();
 }
 
+ss::shard_id connection_cache::shard_for(
+  model::node_id src_node,
+  ss::shard_id src_shard,
+  model::node_id dest_node,
+  ss::shard_id total_shards) const {
+    if (total_shards <= _max_connections) {
+        return src_shard;
+    }
+
+    // NOLINTNEXTLINE
+    uint64_t connection_hash = 201326611; // magic prime
+    boost::hash_combine(connection_hash, std::hash<model::node_id>{}(src_node));
+    boost::hash_combine(
+      connection_hash, std::hash<model::node_id>{}(dest_node));
```

Comment:
`absel::HashOf` is a more concise way to hash several values of presumably similar quality.

Reply:
(though I see now this comes from the existing mapping function which had moved files)